cluster.tf

```diff
@@ -0,0 +1,42 @@
+# GKE cluster
+resource "google_container_cluster" "primary" {
+  name     = local.resource_name
+  location = data.google_compute_zones.available.region
+
+  remove_default_node_pool = true
+  initial_node_count       = 1
+  network                  = data.ns_connection.network.outputs.vpc_name
+  subnetwork               = data.ns_connection.network.outputs.public_subnet_names[0]
+}
+
+# Service account to manage permission via IAM
+resource "google_service_account" "default" {
+  account_id   = local.resource_name
+  display_name = "${local.resource_name} Service Account"
+}
+
+# Managed Node Pool
+resource "google_container_node_pool" "primary_nodes" {
+  name     = "${google_container_cluster.primary.name}-node-pool"
+  location = data.google_compute_zones.available.region
+  cluster  = google_container_cluster.primary.name
+
+  node_count = var.node_count
+
+  node_config {
+    machine_type    = var.node_machine_type
+    service_account = google_service_account.default.email
+    oauth_scopes = [
+      "https://www.googleapis.com/auth/cloud-platform"
+    ]
+
+    labels = {
+      env = data.google_compute_zones.available.project
+    }
+
+    tags = ["gke-node", "${data.google_compute_zones.available.project}-gke"]
```

Comment:
What's the difference between tags and labels

Reply:
Tags are [GCP network tag](https://cloud.google.com/vpc/docs/add-remove-network-tags). Used with firewall rules and route.
[Labels](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/) are Kubernetes node labels that can be used in node selectors to control how workloads are scheduled to your nodes

Reply:
<img width="580" alt="Screenshot 2021-07-28 at 7 44 34 PM" src="https://user-images.githubusercontent.com/5638929/127337937-16f08dc6-df36-420d-bb90-136afcd51161.png">

Reply:
Ah, nice. This is great!